sympy/sets/fancysets.py

```diff
@@ -766,6 +766,15 @@ def _sup(self):
     def _boundary(self):
         return self
 
+    def as_relational(self, symbol):
+        """Rewrite a FiniteSet in terms of equalities and logic operators. """
+        inf_end_points = [S.NegativeInfinity, S.Infinity]
+        if self.start in inf_end_points or self.stop in inf_end_points:
```

Comment:
Perhaps `if self.start.is_infinite or self.stop.is_finite:`?

Reply:
Can `is_infinite` return `None`?

Reply:
```python
>>> Range(1,oo)
Range(1, oo, 1)
>>> oo.is_Integer
False
```

Reply:
> Can `is_infinite` return `None`?

All boolean assumptions return `None` if it's not possible to determine whether the condition is `True` or `False`.